hoymiles_modbus/client.py

```diff
@@ -4,7 +4,9 @@
 from typing import List, Type, Union
 
 from pymodbus.client import ModbusTcpClient
-from pymodbus.framer.socket_framer import ModbusSocketFramer
+from pymodbus.factory import ClientDecoder
+#from pymodbus.framer.socket import FramerSocket
```

Comment:
Have you tried the new framer? Is it working? If yes would prefer using it instead of the legacy, to not rework this again in the future.

Reply:
Baby steps.
Swapping from legacy to new is a major task of rewriting the entire thing. I did this to get the version bumped. When everybody is agreeing this to work, it will be easier to move the upgrade further.